src/lint.cpp

```diff
@@ -247,6 +249,12 @@ void linter::visit_exit_block_scope() {
   this->scopes_.pop();
 }
 
+void linter::visit_exit_with_scope() {
+  QLJS_ASSERT(!this->scopes_.empty());
+  this->propagate_variable_declarations_to_parent_scope();
```

Comment:
Nit: Maybe this deserves a comment? `// Don't propagate variable uses, only declarations.`

Reply:
yes, done